### PR TITLE
Player Polls Revamp

### PR DESCRIPTION
**Pull request by monster860**
# What's wrong with the old polls system?

- It was designed around /tg/station's website. This means that when /tg/ coders added the poll system, they pretty much boned every single downstream server in existence by not making the entire interface in-game.
- The process for creating a poll is somewhat confusing. First of all, you need to find the button placed in the Server tab, which is called "Create Server Poll", and then you need to enter the data in various default byond popups, where you can't see the entire poll all at once.
- The fact that there is a poll is somewhat easy to skim over. It does stick out a bit, but it isn't in your face at all.
- Even if you do click "Show Player Polls", all you get is a window with a single link in grey text, which is not efficient at all, since you have to click on that to get to the actual poll. Even worse if you have multiple polls running, since you'd have to remember which ones you already voted on or check each one.
- It is very easy to cheat the polls. Nothing prevents people from creating multiple polls, and the system is vulnerable to HREF exploits allowing non-admins to vote on admin-only polls.

# How does this PR fix these issues
- Accessing the results of polls now happens entirely in an in-game interface. The results windows are based off of [/tg/'s web-portal for accessing poll results](http://www.ss13.eu/tgdb/tg/ingamepolls.php?page=1).
- Polls are now created using a window which shows you all the information you will need to enter all at once.
- When you join the server, if there is an active poll the window automatically pops up.
- If you have not voted on a certain question, the question and the polling form will be shown inside the polling window
- Your account now needs to be at least 30 days old in order to vote, and HREF exploits have been patched

# Screenshots

# Changelogs
:cl: monster860
add: Adds an in-game way of viewing poll results
tweak: The player poll window automatically pops up if there is an active poll you haven't voted on.
add: Adds an browser popup for creating new polls
tweak: Your account needs to be at least 30 days old in order to vote on polls
/:cl:

### CODE REVIEW

**Comment by Necaladun** [comment]
This looks like it could be super useful and also fun. Yay!

File: code/modules/mob/new_player/login.dm [hunk 1]
```diff
@@ -30,6 +30,23 @@
 	callHook("mob_login", list("client" = client, "mob" = src))
 
 	new_player_panel()
+	
+	spawn(30)
+		// Annoy the player with polls.
+		establish_db_connection()
+		if(dbcon.IsConnected() && client.can_vote())
```
Comment (Crazylemon64) [inline code/modules/mob/new_player/login.dm]: `log_client_to_db` runs after `mob/Login` - so this first `can_vote` will runtime, as `player_age` is `"--"`

**Comment by Crazylemon64** [comment]
sorry for taking forever to do an overview

File: code/modules/admin/create_poll.dm [hunk 2]
```diff
@@ -1,162 +1,202 @@
...
+	var/starttime
+	var/endtime
+	var/DBQuery/query = dbcon.NewQuery("SELECT Now() AS starttime, ADDDATE(Now(), INTERVAL [poll_len] DAY) AS endtime")
+	query.Execute()
+	while(query.NextRow())
+		starttime = query.item[1]
+		endtime = query.item[2]
+	
+	var/pollquery = "INSERT INTO [format_table_name("poll_question")] (polltype, starttime, endtime, question, adminonly, multiplechoiceoptions, createdby_ckey, createdby_ip) VALUES ('[polltype]', '[starttime]', '[endtime]', '[question]', '[adminonly]', '[choice_amount]', '[sql_ckey]', '[address]')"
+	var/idquery = "SELECT id FROM [format_table_name("poll_question")] WHERE question = '[question]' AND starttime = '[starttime]' AND endtime = '[endtime]' AND createdby_ckey = '[sql_ckey]' AND createdby_ip = '[address]'"
```
Comment (monster860) [inline code/modules/admin/create_poll.dm]: It's sanitized on https://github.com/ParadiseSS13/Paradise/pull/6596/files#diff-463efed0062b39a85edb29fa9f52ef6cL90 though...

File: code/modules/mob/new_player/login.dm (hunk 1 above)
Comment (monster860) [inline code/modules/mob/new_player/login.dm]: That's what the `spawn(30)` is for.

**Comment by Fox-McCloud** [comment]
@monster860

**Comment by Fox-McCloud** [comment]
Going to give you until Saturday/Sunday to address this, then this is getting closed. Please update.

**Comment by monster860** [comment]
Guess it's time to suffer through another 20-minute compile/run cycle

**Comment by monster860** [comment]
Alright it should be resolved now. However, the input I was talking about is sanitized, it appears that Crazylemon managed to skim over the part that sanitizes it.

**Comment by Fox-McCloud** [comment]
@Crazylemon64

**Pull request merged**